### DISCUSSION TITLE
Trying to create Internet container above in the diagram to AWS and GCP blocks underneath it

### DISCUSSION THREAD

**Discussion by HariSekhon** [terrastruct/d2#1416] [General]
I'm struggling with the layout of the following diagram which I'm trying to get the Internet section to be cleanly above the AWS and GCP sections in a triangle type shape, but so far all my intermediate/blank objects tricks have failed:

![kubernetes_integrations](https://github.com/terrastruct/d2/assets/2211051/cafadb51-416c-4c6b-85bd-8b5cc993af7a)

As you can see the current diagram is a mess, here is the source code:

https://github.com/HariSekhon/Diagrams-as-Code/blob/master/custom/kubernetes_integrations.d2

I have tried the usual sort of intermediate / blank tricks, such as :

```d2
internet -> intermediate
intermediate -> aws
intermediate -> gcp
```
but this doesn't help as the intermediate object just gets put in between the AWS and GCP blocks:

![kubernetes_integrations](https://github.com/terrastruct/d2/assets/2211051/a694f71e-66eb-4ba1-9836-0a5662e15974)

I think this comes down to the lack of placement control, which is my single biggest with for D2.

I can't use `near` placement control due to this issue of not being able to connect objects: https://github.com/terrastruct/d2/issues/1285

Relates to this issue too: https://github.com/terrastruct/d2/discussions/1300

Anybody got any ideas of how to work around this or any tweaks for now?

**Comment by alixander** [comment]
alright i have some ideas for hacking this in dagre's layout engine that don't involve any user-side dsl-changing hacks, hopefully will get time to try this week.

in the meantime i'm curious if you're able to get the results you're hoping for from mermaid or python Diagrams or any other. I suspect not but don't know fo rsure. No worries if it's too time consuming to replicate.
